indico/modules/events/timetable/controllers/display.py

```diff
@@ -26,12 +27,20 @@
 from indico.web.util import jsonify_data, jsonify_template
 
 
-class RHTimetable(RHDisplayEventBase):
+class RHTimetableProtectionBase(RHDisplayEventBase):
+    def _check_access(self):
+        RHDisplayEventBase._check_access(self)
+        published = contribution_settings.get(self.event, 'published')
+        if not published:
+            raise Forbidden
```

Comment:
shouldn't this use a proper message as well? also, you're using NotFound in some places and Forbidden in others..